Make failed in Windows

cmake succeeded:
<pre>$ cmake -DCMAKE_C_COMPILER=/mingw64/bin/gcc.exe -DCMAKE_CXX_COMPILER=/mingw64/bin/g++.exe .. -G "MSYS Makefiles"
-- The C compiler identification is GNU 5.2.0
-- The CXX compiler identification is GNU 5.2.0
-- Check for working C compiler: D:/Environment/Msys2/mingw64/bin/gcc.exe
-- Check for working C compiler: D:/Environment/Msys2/mingw64/bin/gcc.exe -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: D:/Environment/Msys2/mingw64/bin/g++.exe
-- Check for working CXX compiler: D:/Environment/Msys2/mingw64/bin/g++.exe -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found PROTOBUF: D:/Environment/Msys2/usr/lib/libprotobuf.dll.a
-- Found PythonInterp: D:/Environment/Msys2/usr/bin/python3.4.exe (found suitable version "3.4.3", minimum required is "3.4.0")
-- Configuring done
-- Generating done
-- Build files have been written to: D:/Workspace/Vagrant/libArcus/bui</pre>

Make failed:  
<pre>$ make
Scanning dependencies of target Arcus
[ 12%] Building CXX object CMakeFiles/Arcus.dir/src/Socket.cpp.obj
In file included from D:/Workspace/Vagrant/libArcus/src/Socket.cpp:20:0:
D:/Workspace/Vagrant/libArcus/src/Socket_p.h: In member function 'void Arcus::SocketPrivate::error(std::__cxx11::string)':
D:/Workspace/Vagrant/libArcus/src/Socket_p.h:150:23: error: 'socketId' was not declared in this scope
         ::closesocket(socketId);
                       ^
D:/Workspace/Vagrant/libArcus/src/Socket_p.h: In member function 'void Arcus::SocketPrivate::run()':
D:/Workspace/Vagrant/libArcus/src/Socket_p.h:227:35: error: 'socketId' was not declared in this scope
                     ::closesocket(socketId);
                                   ^
D:/Workspace/Vagrant/libArcus/src/Socket_p.h:278:35: error: 'socketId' was not declared in this scope
                     ::closesocket(socketId);
                                   ^
CMakeFiles/Arcus.dir/build.make:62: recipe for target 'CMakeFiles/Arcus.dir/src/Socket.cpp.obj' failed
make[2]: *** [CMakeFiles/Arcus.dir/src/Socket.cpp.obj] Error 1
CMakeFiles/Makefile2:67: recipe for target 'CMakeFiles/Arcus.dir/all' failed
make[1]: *** [CMakeFiles/Arcus.dir/all] Error 2
Makefile:149: recipe for target 'all' failed
make: *** [all] Error 2
</pre>

I tried to fix this by changing "socketId" to "socket_id",then I got a new error:
<pre>$ make
[ 12%] Linking CXX shared library libArcus.dll
CMakeFiles/Arcus.dir/objects.a(Socket.cpp.obj):Socket.cpp:(.text+0xe98): undefined reference to `google::protobuf::MessageLite::SerializeAsString[abi:cxx11]() const'
collect2.exe: error: ld returned 1 exit status
CMakeFiles/Arcus.dir/build.make:120: recipe for target 'libArcus.dll' failed
make[2]: *** [libArcus.dll] Error 1
CMakeFiles/Makefile2:67: recipe for target 'CMakeFiles/Arcus.dir/all' failed
make[1]: *** [CMakeFiles/Arcus.dir/all] Error 2
Makefile:149: recipe for target 'all' failed
make: *** [all] Error 2
</pre>

--- a/src/Socket_p.h
+++ b/src/Socket_p.h
@@ -147,7 +147,7 @@ namespace Arcus
     {
     	error_string = msg;
 #ifdef _WIN32
-        ::closesocket(socketId);
+        ::closesocket(socket_id);
 #else
         ::close(socket_id);
 #endif
@@ -224,7 +224,7 @@ namespace Arcus
                     }
 
                 #ifdef _WIN32
-                    ::closesocket(socketId);
+                    ::closesocket(socket_id);
                 #else
                     ::close(socket_id);
                 #endif
@@ -275,7 +275,7 @@ namespace Arcus
                     }
 
                 #ifdef _WIN32
-                    ::closesocket(socketId);
+                    ::closesocket(socket_id);
                 #else
                     ::close(socket_id);
                 #endif

